content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to your models, result objects, queries, or to Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To use a Prisma Client extension, you must create an associated _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. The standard client does not change.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). You can run an extension in its own isolated extended client, or combine it with other extensions in an extended client.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+### Client instances
+
+Extended clients interact with each other, and with the standard client, as follows:
+
+- Each extended client operates independently in an isolated instance.
```

Comment:
@SevInf what is your opinion on this one? I'd say this is not so true anymore since the introduction of `Prisma.getExtensionContext`.

Reply:
@SevInf - what do you think about this?

Reply:
@SevInf - what do you think about this?

Reply:
I read this sentence as follows: if we have two instances, `const xclient1 = prisma.$extends(ext1)` and `const xclient2 = prisma.$extends(ext2)` nothing `ext1` does will affect `xclient2` and nothing `ext2` does will affect `xclient1`. That is still true in this case. 

On other hand, if we do this:

```
const xclient = prisma.$extends(ext1).$extends(ext2)
```

Then yes, `ext2` can depend on `ext1`.

Reply:
Thanks. That was my understanding as well. I'll review this section to make sure it's clearer, because if we're misunderstanding it, then users will

Reply:
@SevInf  I think we are confusing extended clients and extensions here. If you have two separate **extended clients**, then I think they are always isolated. But in your second example, these are two **extensions** in the same **extended client** and they are not isolated from each other.

The terminology we've chosen isn't helping here. I have proposed that we come up with a new term to describe the forked client instances that doesn't have some variation of the word "extend" in it.